sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryUtils.java

```diff
@@ -82,11 +84,20 @@
 
                 if (!shouldRetryResult.shouldRetry) {
                     retryPolicy.updateEndTime();
-                    if(shouldRetryResult.exception == null) {
-                        return Mono.error(e);
-                    } else {
-                        return Mono.error(shouldRetryResult.exception);
+
+                    final Throwable errorToReturn = shouldRetryResult.exception != null ? shouldRetryResult.exception : e;
+                    final Mono<T> failure = Mono.error(errorToReturn);
+
+                    if (shouldRetryResult.policyArg != null &&
+                        shouldRetryResult.policyArg.getValue0() != null &&
+                        shouldRetryResult.policyArg.getValue0()) {
+
+                        return addressSelector.resolveAddressesAsync(
```

Comment:
this is for refreshing address cache on a Gone Exception for write request.

With this approach the write response will get blocked till the unrelated address resolution completes.

you could instead trigger a address cache refresh in background and return the write response immediately.

See this as example:
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java#L365-L373

Reply:
Fixed - good catch. Thanks!